src/coreclr/jit/morph.cpp

```diff
@@ -935,10 +935,20 @@ void CallArgs::ArgsComplete(Compiler* comp, GenTreeCall* call)
 #if defined(FEATURE_SIMD) && defined(TARGET_ARM64)
                 else if (isMultiRegArg && varTypeIsSIMD(argx->TypeGet()))
                 {
+                    GenTree* nodeToCheck = argx;
+
+                    if (nodeToCheck->OperIs(GT_OBJ))
+                    {
+                        nodeToCheck = nodeToCheck->AsObj()->gtOp1;
+
+                        if (nodeToCheck->OperIs(GT_ADDR))
+                        {
+                            nodeToCheck = nodeToCheck->AsOp()->gtOp1;
+                        }
+                    }
+
                     // SIMD types do not need the optimization below due to their sizes
-                    if (argx->OperIsSimdOrHWintrinsic() ||
-                        (argx->OperIs(GT_OBJ) && argx->AsObj()->gtOp1->OperIs(GT_ADDR) &&
-                         argx->AsObj()->gtOp1->AsOp()->gtOp1->OperIsSimdOrHWintrinsic()))
+                    if (nodeToCheck->OperIsSimdOrHWintrinsic() || nodeToCheck->IsCnsVec())
```

Comment:
Why is this needed given in `impNormStructVal` we don't actually create `OBJ(ADDR(CNS_VEC))` (which is a good thing), but use a temp instead?

Reply:
`makeTemp` doesn't avoid `OBJ(ADDR(x))` it just ensures its `OBJ(ADDR(LCL_VAR))` (assuming the value isn't a `blk` operation).

This was a limitation around handles and ensuring it could be tracked where necessary for call nodes and the like. It's a more complex change to remove

Reply:
> it just ensures its `OBJ(ADDR(LCL_VAR))`

Indeed, so my question was why do we need to change this code, seeing as we do not have `OBJ(ADDR(CNS_VEC))`.

Reply:
Fixed.